File: jokenpo/2018-1/elisson ruby/jokenpo.rb
class Jokenpo
  def initialize(opcao_1, opcao_2)
    @opcao_1 = opcao_1
    @opcao_2 = opcao_2
  end

  def vencedor
    return "Empate" if empate
    if @opcao_1.vence(@opcao_2)
      jogador_1_venceu
    else
      jogador_2_venceu
    end
  end

  private
  def empate
    @opcao_1.class == @opcao_2.class
  end

  def jogador_1_venceu
    "Jogador 1 Venceu. " +
    "#{@opcao_1.class} Vence #{@opcao_2.class}"
  end

  def jogador_2_venceu
    "Jogador 2 Venceu. " +
    "#{@opcao_2.class} Vence #{@opcao_1.class}"
  end
end

File: #29 jokenpo/2018-1/elisson java/Jokenpo.java
public class Jokenpo {
	public Opcao opcao1;
	public Opcao opcao2;
	
	Jokenpo(Opcao opcao1, Opcao opcao2){
		this.opcao1 = opcao1;
		this.opcao2 = opcao2;
	}
	
	private boolean empate() {
		return !opcao1.vence(opcao2) && !opcao2.vence(opcao1);
	}
	
	public String vencedor() {
		if(empate()) return "Empate";
		if(opcao1.vence(opcao2)) {
			return "Jogador 1 Venceu";
		} else {
			return "Jogador 2 Venceu";
		}
	}
}
